internal/dag/conditions.go

```diff
@@ -59,6 +66,32 @@ func pathConditionsValid(sw *ObjectStatusWriter, conds []projcontour.Condition,
 				sw.SetInvalid(fmt.Sprintf("%s: Prefix conditions must start with /, %s was supplied", conditionsContext, cond.Prefix))
 				return false
 			}
+			switch conditionsContext {
+			case "include":
+				if strings.Contains(cond.Prefix, "*") {
+					sw.SetInvalid(fmt.Sprintf("Cannot specify wildcard prefix conditions in an Include"))
+					return false
+				}
+			case "route":
+				if strings.HasSuffix(cond.Prefix, "*") {
+					sw.SetInvalid(fmt.Sprintf("Cannot specify trailing wilcard character, '%s' was supplied", cond.Prefix))
+					return false
+				}
+				if strings.HasPrefix(cond.Prefix, "/*") {
+					sw.SetInvalid(fmt.Sprintf("Cannot specify '/*' as leading wilcard character pattern, '%s' was supplied", cond.Prefix))
+					return false
+				}
+				if strings.Contains(cond.Prefix, "**") {
+					sw.SetInvalid(fmt.Sprintf("Cannot specify '**' character pattern in a wildcard prefix, '%s' was supplied", cond.Prefix))
+					return false
+				}
+				for _, s := range strings.Split(cond.Prefix, "/") {
+					if strings.Count(s, "*") > 1 {
+						sw.SetInvalid(fmt.Sprintf("Cannot specify multiple '*' between slashes in a wildcard prefix, '%s' was supplied", cond.Prefix))
+						return false
+					}
+				}
```

Comment:
This might be cleaner if you split first, then apply the rules to the resulting slice.

Reply:
Can you elaborate a bit more? Not sure I'm following you.

Reply:
If you start the validation by splitting the path into a slice of components, then you can apply all the checks component-wise after that. I think the whole thing will simplify.